src/inc/quic_platform_posix.h

```diff
@@ -470,9 +474,11 @@ CxPlatPoolAlloc(
     _Inout_ CXPLAT_POOL* Pool
     )
 {
-#if QUIC_DISABLE_MEM_POOL
-    return CxPlatAlloc(Pool->Size);
-#else
+#if DEBUG
+    if (CxPlatGetDisablePoolAllocator()) {
```

Comment:
Instead of making a separate flag, let's just disable pool alloc if any alloc failure is enabled.